src/pkgcheck/checks/metadata.py

```diff
@@ -282,6 +296,11 @@ def feed(self, pkg):
         elif eapi_str in self.options.target_repo.config.eapis_deprecated:
             yield DeprecatedEapi(pkg.eapi, pkg=pkg)
 
+        if eapi_str in self.options.target_repo.config.eapis_testing:
+            stable_keywords = [k for k in pkg.keywords if not k.startswith(('~', '-'))]
+            if stable_keywords:
+                yield StableKeywordsOnTestingEapi(pkg.eapi, stable_keywords, pkg=pkg)
```

Comment:
So in this case you'd do something like:

```python
if stable_keywords := sorted(k for k in pkg.keywords if not k.startswith(('~', '-'))):
```

and then

```python
self.keywords = tuple(keywords)
```

in the StableKeywordsOnTestingEapi class.

Reply:
I will also note that pkgcheck generally uses assignment expressions where it makes sense, so if you want to keep internal usage consistent you'd use one here.

Reply:
Wouldn't it exceed line width though?

Reply:
> Wouldn't it exceed line width though?

I didn't check, but pkgcheck et al. use a general loose standard of 100 chars so if exceeds that, I guess it's fine.

Reply:
Having checked, nope it's 98 chars when using the assignment op. :laughing:

Reply:
Anyway, I'd generally split it into a separate generator var, and then use the assignment operator along with sorted() which wouldn't even get close to 100 chars.

Reply:
I'm sorry, it's just been a long day.